### DISCUSSION TITLE
Implementing custom authenticator (extension) in java , and configure it in otel java instrumentation

### DISCUSSION THREAD

**Discussion by shadow0wolf** [open-telemetry/opentelemetry-java#5993] [Q&A]
I want to implement a custom client authenticator extension in otel-java-sdk , i have done this as a serverAuthenticator  in golang on otel collector but i am not able to find any documentation on how to do this for otl-java-sdk .

**Comment by jkwatson** [comment]
I don't think we have any specification about what we need to support in the core libraries in this regard. Do you know if there is a specification requirement to provide extension points on exporters? [I assume this is related to exporters authenticating to the collector. please correct if I'm wrong].

**Reply by shadow0wolf** [reply]
I dont have any formal specification but something analogous to this : https://opentelemetry.io/docs/collector/custom-auth/ (client authenticator)

**Comment by trask** [comment]
hi @shadow0wolf, is this the same as https://github.com/open-telemetry/opentelemetry-java/issues/5959? thx

**Reply by shadow0wolf** [reply]
NO , but a generic question regarding implementing custom Authenticator as an extension on java instrumented .
